bigbio/utils/license.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+License objects.
+"""
+
+import importlib.resources as pkg_resources
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from bigbio.utils import resources
+
+
+@dataclass
+class License:
+    """
+    Base class from which all licenses inherit
+    """
+
+    name: Optional[str] = None
+    text: Optional[str] = None
+    link: Optional[str] = None
+    version: Optional[float] = None
```

Comment:
even if its not always filled out, I was thinking it would be nice to have a `provenance` field (or some name like that) that could be used to indicate the source of the license attribution (i.e. where can someone look to confirm this dataset has this license)